front/components/vaults/ContentActions.tsx

```diff
@@ -0,0 +1,154 @@
+import { PencilSquareIcon, TrashIcon } from "@dust-tt/sparkle";
+import type {
+  DataSourceViewType,
+  LightContentNode,
+  PlanType,
+  WorkspaceType,
+} from "@dust-tt/types";
+import { isFolder, isWebsite } from "@dust-tt/types";
+import type { RefObject } from "react";
+import React, { useImperativeHandle, useState } from "react";
+
+import { DocumentDeleteDialog } from "@app/components/data_source/DocumentDeleteDialog";
+import { DocumentUploadOrEditModal } from "@app/components/data_source/DocumentUploadOrEditModal";
+import { MultipleDocumentsUpload } from "@app/components/data_source/MultipleDocumentsUpload";
+import { TableDeleteDialog } from "@app/components/data_source/TableDeleteDialog";
+import { TableUploadOrEditModal } from "@app/components/data_source/TableUploadOrEditModal";
+
+type ContentActionKey =
+  | "DocumentUploadOrEditModal"
+  | "MultipleDocumentsUpload"
+  | "DocumentDeleteDialog"
+  | "TableUploadOrEditModal"
+  | "TableDeleteDialog";
+
+export type ContentAction = {
+  action?: ContentActionKey;
+  contentNode?: LightContentNode;
+};
+
+type ContentActionsProps = {
+  dataSourceView: DataSourceViewType;
+  plan: PlanType;
+  owner: WorkspaceType;
+  onSave: () => void;
+};
+
+export type ContentActionsRef = {
+  callAction: (
+    action: ContentActionKey,
+    contentNode?: LightContentNode
+  ) => void;
+};
+
+export const ContentActions = React.forwardRef<
+  ContentActionsRef,
+  ContentActionsProps
+>(({ dataSourceView, owner, plan, onSave }: ContentActionsProps, ref) => {
+  const [currentAction, setCurrentAction] = useState<ContentAction>({});
+  useImperativeHandle(ref, () => ({
+    callAction: (action: ContentActionKey, contentNode?: LightContentNode) => {
```

Comment:
Would go with `openModalForAction`.

Reply:
This could actually do other actions than opening modal (like download content, for example)